SchoolProject/SchoolApp/views.py

```diff
@@ -1,85 +1,262 @@
-def subject (request):
-    subs = Subject.objects.all()
-    return  render(request,'subject.html',{'subs':subs})
+class subject (View):
+    def get(self,request,*args, **kwargs):
+        subs = Subject.objects.all()
+        return render(request, 'subject.html', {'subs': subs})
 
-def classRoom (request):
-    cRooms = ClassRoom.objects.all()
-    return  render(request,'classRoom.html',{'cRooms':cRooms})
+class classRoom (View):
+    def get(self,request,*args, **kwargs):
+        cRooms = ClassRoom.objects.all()
+        return render(request, 'classRoom.html', {'cRooms': cRooms})
 
-def grade (request):
-    grds = Grade.objects.all()
-    return  render(request,'grade.html',{'grds':grds})
+class grade (View):
+    def get(self,request,*args, **kwargs):
+        grds = Grade.objects.all()
+        return render(request, 'grade.html', {'grds': grds})
 
 # For Add/Update entities
-def studentForm (request):
-    studs = Student.objects.all()
-    subs = Subject.objects.all()
-    grds = Grade.objects.all()
-    context = {'studs': studs, 'subs': subs, 'grds': grds}
-    if request.method == "POST":
-        studentInfo = Student.objects.create (studentName=request.POST.get('StudentName'), birthdate=request.POST.get('Birthdate'), studentSubjects=request.POST.get('StudentSubject'), studentGrade=request.POST.get('StudentGrade'))
-    return  render(request,'studentForm.html',context)
-
-def teacherForm (request):
-    subs = Subject.objects.all()
-    if request.method == "POST":
-       teacherInfo = Teacher.objects.create (teacherName=request.POST.get('TeacherName'), teacherSubjects=request.POST.get('TeacherSubjects'))
-    return  render(request,'teacherForm.html', {'subs':subs})
-
-def subjectForm (request):
-    clsRoom = ClassRoom.objects.all()
-    if request.method == "POST":
-      subjectInfo = Subject.objects.create (subjectName=request.POST.get('SubjectName'),classRoomName=request.POST.get('ClassRoomName'))
-      print(request.POST.get('ClassRoomName'))
-
-    return  render(request,'subjectForm.html',{'clsRoom':clsRoom})
-
-def classRoomForm (request):
-    if request.method == "POST":
-      roomName = ClassRoom.objects.create(classRoomName=request.POST.get('RoomName'))
-    return render(request, 'classRoomForm.html')
-
-def gradeForm (request):
-    if request.method == "POST":
-        if request.POST.get('AddGrade'):
-           gradeName = Grade.objects.create(gradeName=request.POST.get('GradeName'))
-           return render(request, 'gradeForm.html')
-        elif request.POST.get('UpdateGrade'):
-              grdName = Grade.objects.get(gradeName='UpdateGradeName')
-              form = gradeForms(request.POST,instance=grdName)
-              if form.is_valid():
-                  form.save()
-                  messages.success(request,"Grade Updated Successfully")
-                  return render(request, 'gradeForm.html', {'grdName': grdName})
-    return render(request, 'gradeForm.html')
+class studentForm(View):
+
```

Comment:
u said in  teams that in class view It reduce code duplication , 
but I see same amount of code is used here , 
take a second look please on how to use the class based views in Django